lib/Models/Catalog.js

```diff
@@ -149,30 +125,49 @@ Catalog.prototype.updateFromJson = function(json, options) {
  *
  * @param {Object} [options] Object with the following properties:
- * @param {Boolean} [options.enabledItemsOnly=false] true if only enabled data items (and their groups) should be serialized,
- *                  or false if all data items should be serialized.
- * @param {CatalogMember[]} [options.itemsSkippedBecauseTheyAreNotEnabled] An array that, if provided, is populated on return with
- *        all of the data items that were not serialized because they were not enabled.  The array will be empty if
- *        options.enabledItemsOnly is false.
- * @param {Boolean} [options.skipItemsWithLocalData=false] true if items with a serializable 'data' property should be skipped entirely.
- *                  This is useful to avoid creating a JSON data structure with potentially very large embedded data.
- * @param {CatalogMember[]} [options.itemsSkippedBecauseTheyHaveLocalData] An array that, if provided, is populated on return
- *        with all of the data items that were not serialized because they have a serializable 'data' property.  The array will be empty
- *        if options.skipItemsWithLocalData is false.
- * @param {Boolean} [options.serializeForSharing=false] true to only serialize properties that are typically necessary for sharing this member
- *                                                      with other users, such as {@link CatalogGroup#isOpen}, {@link CatalogItem#isEnabled},
- *                                                      {@link CatalogItem#isLegendVisible}, and {@link ImageryLayerCatalogItem#opacity},
- *                                                      rather than serializing all properties needed to completely recreate the catalog.
- * @param {Boolean} [options.userSuppliedOnly=false] true to only serialize catalog members (and their containing groups) that have been identified as having been
- *                  supplied by the user ({@link CatalogMember#isUserSupplied} is true); false to serialize all catalog members.
+ * @param {Function} [options.propertyFilter] Filter function that will be executed to determine whether a property
+ *          should be serialized.
+ * @param {Function} [options.itemFilter] Filter function that will be executed for each item in a group to determine
+ *          whether that item should be serialized.
  * @return {Object} The serialized JSON object-literal.
  */
 Catalog.prototype.serializeToJson = function(options) {
-     this.terria.nowViewing.recordNowViewingIndices();
+    this.terria.nowViewing.recordNowViewingIndices();
+
+    return this.group.serializeToJson(options).items;
+};
 
-    var json = {};
-    CatalogGroup.defaultSerializers.items(this.group, json, 'items', options);
-    return json.items;
+/**
+ * Resolves items in the catalog based on the share keys provided, and enabled them along with all their ancestors in
+ * the catalog hierarchy. This is asynchronous, as to enable an item it must have finished loading first.
```

Comment:
Yeah, I need the promise so I can wait for this to load before trying to enable its children (because otherwise I won't know whether they exist to match keys with). Probably can just immediately enable after calling `load` then.